orc8r/cloud/go/services/orc8r_worker/orc8r_worker/main.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"magma/orc8r/cloud/go/services/orc8r_worker"
+
+	"github.com/golang/glog"
+
+	"magma/orc8r/cloud/go/blobstore"
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/service"
+	"magma/orc8r/cloud/go/services/state"
+	state_config "magma/orc8r/cloud/go/services/state/config"
+	"magma/orc8r/cloud/go/services/state/indexer/reindex"
+	"magma/orc8r/cloud/go/services/state/metrics"
+	indexer_protos "magma/orc8r/cloud/go/services/state/protos"
+	"magma/orc8r/cloud/go/services/state/servicers"
+	"magma/orc8r/cloud/go/sqorc"
+	"magma/orc8r/cloud/go/storage"
+	"magma/orc8r/lib/go/service/config"
+)
+
+// how often to report gateway status
+const gatewayStatusReportInterval = time.Second * 60
+
+func main() {
+	srv, err := service.NewOrchestratorService(orc8r.ModuleName, orc8r_worker.ServiceName)
+	if err != nil {
+		glog.Fatalf("Error creating orc8r_worker service %v", err)
+	}
+
+	singletonReindex := srv.Config.MustGetBool(state_config.EnableSingletonReindex)
+	if singletonReindex {
+		glog.Info("Running singleton reindexer")
+
+		db, err := sqorc.Open(storage.GetSQLDriver(), storage.GetDatabaseSource())
+		if err != nil {
+			glog.Fatalf("Error connecting to database: %v", err)
+		}
+		store := blobstore.NewSQLStoreFactory(state.DBTableName, db, sqorc.GetSqlBuilder())
+		err = store.InitializeFactory()
+		if err != nil {
+			glog.Fatalf("Error initializing state database: %v", err)
+		}
+
+		indexerManagerServer := newSingletonIndexerManagerServicer(srv.Config, db, store)
+		indexer_protos.RegisterIndexerManagerServer(srv.GrpcServer, indexerManagerServer)
+	}
```

Comment:
Since the `orc8r_worker` is generic (any number of singleton jobs), let's move this block into its own `startReindexer` func

Reply:
it's the entire indexer if that's ok?

Reply:
Maybe I'm misreading -- I don't see where it's adding an indexer?

Reply:
https://github.com/magma/magma/pull/10172/files#diff-5cd2a79f7253b049992f66579be1a14a4cd5a9360c978afefdb10d92b745baccR81

(L81 if it doesn't highlight properly)

Reply:
I think it's adding an indexer? Not sure now

Reply:
Ah right. So that's an indexer *manager* servicer, which exposes the grpc servicer used by the (re)index CLI https://github.com/magma/magma/tree/13b485a9b6da5874761919c840e4c0a8be81641d/orc8r/cloud/go/tools/indexers

Reply:
ahh makes sense thanks for the info!